controllers/metallb_controller.go

```diff
@@ -81,10 +82,14 @@ func (r *MetallbReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		// Error reading the object - requeue the request.
 		return ctrl.Result{}, err
 	}
+	status.UpdateStatus(context.TODO(), r.Client, instance, status.ConditionProgressing, "", "")
 
 	err = r.syncMetalLBResources(instance)
 	if err != nil {
+		status.UpdateStatus(context.TODO(), r.Client, instance, status.ConditionDegraded, "FailedToSyncMetalLBResources", err.Error())
```

Comment:
is this the right time to address the question? https://github.com/metallb/metallb-operator/blob/cf404d78b9ed1aabd7e6d43d48e3adf409293ce4/controllers/metallb_controller.go#L103 
since we're taking more actions on errors from syncMetalLBResources should we/should we not return this err? 
https://github.com/metallb/metallb-operator/blob/eee3ab478bfe0834aaf9a68ae9796defb6650f69/controllers/metallb_controller.go#L125
@fedepaol

Reply:
imho we should go over each and return the error to retry
I guess this deserves another pr

Reply:
#29